.github/workflows/CI.yml

```diff
@@ -0,0 +1,66 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - 'v*'
+  pull_request:
+    branches:
+      - main
+
+env:
+  MACOSX_DEPLOYMENT_TARGET: 12.0
+
+jobs:
+  Build:
+    name: Conda-Forge LPython Test ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["windows-2019", "macos-latest", "ubuntu-latest"]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - uses: mamba-org/setup-micromamba@v1
+        with:
+          environment-file: environment.yml
+
+      - uses: hendrikmuhs/ccache-action@main
+        with:
+          variant: sccache
+          key: ${{ github.job }}-${{ matrix.os }}
+
+      - name: Clone LPython Source Repository
+        shell: bash -e -l {0}
+        run: |
+            git clone https://github.com/lcompilers/lpython.git
+            git checkout 2293972e5ce377787d134541f8363ae441030b76
```

Comment:
For easiness of maintenance, perhaps here we can just checkout the corresponding "tag" from `environment.yml`, so:
```
git checkout v0.19.0
```
The git hash is more robust, but here I think we will be updating this version very often, so I think it's fine to use the tag. We assume that we do not change tags once published (we've never done that so far), so it should be robust.

In other cases when we use a branch, that is fragile, since branches move.